### PR TITLE
ProcessVehicleEngine and ProcessActiveQueues

### PR DESCRIPTION
**Pull request by ShFil119**
Also some smaller fixes, thx erorcun for help.

I'll later search for better names for:
```
uint32 v28;
uint32 v29;
```

### CODE REVIEW

File: src/audio/AudioManager.cpp [hunk 1]
```diff
@@ -2742,11 +2743,219 @@ cAudioManager::PreTerminateGameSpecificShutdown()
...
+	for (int32 i = 0; i < m_bSampleRequestQueuesStatus[m_bActiveSampleQueue]; ++i) {
+		tSound& sample = m_asSamples[m_bActiveSampleQueue][m_abSampleQueueIndexTable[m_bActiveSampleQueue][i]];
+		if (sample.m_nSampleIndex != NO_SAMPLE) {
+			for (int32 j = 0; j < m_bActiveSamples; ++j) {
+				if (sample.m_nEntityIndex == m_asActiveSamples[j].m_nEntityIndex &&
+					sample.m_counter == m_asActiveSamples[j].m_counter &&
+					sample.m_nSampleIndex == m_asActiveSamples[j].m_nSampleIndex) {
+					if (sample.m_nLoopCount) {
+						if (m_FrameCounter & 1) {
+							if (!(j & 1)) { flag = 0; }
```
Comment (aap) [inline src/audio/AudioManager.cpp]: maybe just `flag = !!(j & 1);` otherwise there's an else missing.

File: src/audio/AudioManager.cpp [hunk 2]
```diff
@@ -2742,11 +2743,219 @@ cAudioManager::PreTerminateGameSpecificShutdown()
...
+							flag = 1;
+						}
+						else {
+							if (!(j & 1))
```
Comment (aap) [inline src/audio/AudioManager.cpp]: `flag = !(j&1)`

File: src/audio/AudioManager.cpp [hunk 3]
```diff
@@ -2742,11 +2743,219 @@ cAudioManager::PreTerminateGameSpecificShutdown()
...
-WRAPPER
 void
 cAudioManager::ProcessActiveQueues()
 {
-	EAXJMP(0x57BA60);
+	bool flag;
+	float position2;
+	float position1;
+
+	uint32 v28;
```
Comment (erorcun) [inline src/audio/AudioManager.cpp]: you forgot to change those it seems

Comment (ShFil119) [reply src/audio/AudioManager.cpp]: not sure how to name them yet.

**Pull request merged**